M2/Macaulay2/m2/methods.m2

```diff
@@ -683,28 +683,10 @@ protect QuotientRingHook
 -- stashing or caching computed values for future reference in functions that take a mutable hash table as input
 -----------------------------------------------------------------------------
 
-CacheFunction = new Type of FunctionClosure
```

Comment:
Why is CacheFunction going away?  I've never used it, but it looks like maybe a lot of this caching code can be simplified with `??=`?  What above the removing of keys?

Reply:
It's not really used anywhere. `cacheValue` and `stacheValue` returned this type, but my plan is to deprecate those in favor of `??=`.

Reply:
actually, `CacheFunction` is used, for example in the package `LieTypes`, which no longer loads now.

Reply:
if `cacheValue` is to be deprecated, then some discussion is needed on what exactly it will be replaced with.